subprojects/core/src/main/java/org/gradle/api/internal/attributes/DefaultAttributesSchema.java

```diff
@@ -16,63 +16,49 @@
 
 package org.gradle.api.internal.attributes;
 
-import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Maps;
+import org.gradle.api.Action;
 import org.gradle.api.attributes.Attribute;
 import org.gradle.api.attributes.AttributeMatchingStrategy;
-import org.gradle.api.attributes.AttributeValue;
 import org.gradle.api.attributes.AttributesSchema;
 import org.gradle.internal.Cast;
 
-import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
 public class DefaultAttributesSchema implements AttributesSchema {
-    private final Map<Attribute<?>, AttributeMatchingStrategy<?>> strategies = Maps.newHashMap();
 
-    @Override
-    public <T> void setMatchingStrategy(Attribute<T> attribute, AttributeMatchingStrategy<T> strategy) {
-        strategies.put(attribute, strategy);
-    }
+    private final Map<Attribute<?>, AttributeMatchingStrategy<?>> strategies = Maps.newHashMap();
 
     @Override
+    @SuppressWarnings("unchecked")
     public <T> AttributeMatchingStrategy<T> getMatchingStrategy(Attribute<T> attribute) {
         AttributeMatchingStrategy<?> strategy = strategies.get(attribute);
-        if (strategy == null && String.class == attribute.getType()) {
-            strategy = StrictMatchingStrategy.INSTANCE;
-        }
         if (strategy == null) {
             throw new IllegalArgumentException("Unable to find matching strategy for " + attribute);
         }
         return Cast.uncheckedCast(strategy);
     }
 
     @Override
-    public <T> void matchStrictly(Attribute<T> attribute) {
-        setMatchingStrategy(attribute, StrictMatchingStrategy.<T>get());
+    public <T> AttributeMatchingStrategy<T> configureMatchingStrategy(Attribute<T> attribute, Action<? super AttributeMatchingStrategy<T>> configureAction) {
```

Comment:
It looks like someone has to explicitly define each attribute. Possibly we should just default to equality checking for attributes we don't know about.